openfe/protocols/restraint_utils/geometry/boresch/host.py

```diff
@@ -533,6 +548,70 @@ def _conclude(self):
                 self.results.valid[i] = True
 
 
+def _get_lowest_variance_restraint_hostanchor(
+    proposed_restraints: list[mda.AtomGroup],
+    angle_force_constant: Quantity,
+    temperature: Quantity
+) -> list[int] | None:
+    """
+    Evaluate a list of proposed restraints and return the lowest variance
+    valid restraint.
+
+    Parameters
+    ----------
+    proposed_restraints : list[MDAnalysis.AtomGroup]
+      A proposed list of AtomGroup defining the H2-H1-H0-G0-G1-G2
+      restraint atoms, in that order.
+    angle_force_constant : openff.units.Quantity
+      The force constant for the angle.
+    temperature : openff.units.Quanity
+      The system temperature in units compatible with Kelvin.
+
+    Returns
+    -------
+    list[int] | None
+      The H0, H1, H2 host atom indices for the picked restraint, or None
+      if no suitable restraints are found.
+    """
+    # Evaluate all the restraints
+    restraints_eval = EvaluateBoreschAtoms(
+        restraints=proposed_restraints,
+        angle_force_constant=angle_force_constant,
+        temperature=temperature,
+    ).run()
+
+    # If there are no valid restraints, we have nothing
+    if not any(restraints_eval.results.valid):
+        return None
+
+    valid_indices = []
+    valid_variances = []
+    for ridx in range(len(proposed_restraints)):     
+        if restraints_eval.results.valid[ridx]:
+            valid_indices.append(ridx)
+
+            dih_variance = sum([
+                circvar(diheds, high=np.pi, low=-np.pi)
+                for diheds in restraints_eval.results.dihedrals[ridx]
+            ])
+
+            ang_variance = sum([
+                circvar(angles, high=np.pi, low=0)
+                for angles in restraints_eval.results.angles[ridx]
+            ])
+
+            bond_variance = np.var(restraints_eval.results.bonds[ridx])
+
+            valid_variances.append(dih_variance + ang_variance + bond_variance)
```

Comment:
Only using a linear combination of variances here. In the past, this "worked", but probably just as well as any other approaches. Happy to change it to something else if there's a better combination we can try.